---
title: analyze flow cytometry data from 2020-12-10
author: Bin He
date: 2020-12-14
output: 
  html_notebook:
    toc: true
    toc_float: true
    df_print: default
    code_folding: hide
---

```{r setup}
require(tidyverse)
require(cowplot)
```

## Goal

Testing the brightness of Pho4-GFP (mNeon and EGFP) either on a plasmid or integrated at the leu2 or endogenous Pho4 locus. also evaluate the population heterogeneity and use the PHO5pr-mCherry reporter to evaluate the functionality of the Pho4 variants

## Material and methods

### table of strains
```{r echo = F}
samples <- read_tsv("strains-genotype.tsv", col_types = "ccccccccD")
(samples)
```

### methods
See gDoc for details. Some strains didn't grow to saturation after the first overnight culture and thus the o/n culture instead of the second growth, which was at mid log phase at the time of harvest, was used for flow cytometry.

## Data and analysis
```{r}
dat <- read_csv("20201210-testing-Pho4-GFP-on-plasmid-or-integrated-with-PHO5pr-reporter.csv")
dat <- dat %>% mutate(Pho4GFP = factor(Pho4GFP, levels = c("No-GFP","ScPho4-mNeon","ScPho4-EGFP", "CgPho4-mNeon")),
                      Pho4 = factor(Pho4, levels = c("pho4-", "ScPho4", "CgPho4")))
```

### Pho4-GFP expression levels
First look at the level of Pho4-GFP compared with the negative controls.
```{r, echo = F, fig.height = 7, fig.width = 10}
gfp.bg <- dat %>% filter(Pho4GFP == "No-GFP", Parameter == "Pho4-GFP") %>% pull(Median) %>% mean()
p1 <- dat %>% filter(Parameter == "Pho4-GFP", Median < 3000) %>% 
  ggplot(aes(x = Sample, y = Median, fill =  Group)) + 
  geom_bar(stat = "identity", position = position_dodge2(width = 0.9)) +
  geom_text(aes(label = ifelse(Growth == "ovn", "*", ""), hjust = -0.2, vjust = 0.8), 
            size = 5) +
  geom_hline(yintercept = gfp.bg, linetype = "dashed", alpha = 0.6) +
  facet_grid(Pho4GFP~., scales = "free_y", space = "free_y") +
  scale_fill_brewer(type = "div") + ylab("MFI") +
  coord_flip() + scale_y_continuous(limits = c(0,3000), expand = c(0.01,0)) +
  theme_bw() + theme(legend.position = "none")

# include the outliers
p2 <- dat %>% filter(Parameter == "Pho4-GFP") %>% 
  ggplot(aes(x = Sample, y = Median, fill =  Group)) + 
  geom_bar(stat = "identity", position = position_dodge2(width = 0.9)) +
  geom_text(aes(label = ifelse(Growth == "ovn", "*", ""), hjust = -0.2, vjust = 0.8), 
            size = 5) +
  geom_hline(yintercept = gfp.bg, linetype = "dashed", alpha = 0.6) +
  facet_grid(Pho4GFP~., scales = "free_y", space = "free_y") +
  scale_fill_brewer(type = "div") + ylab("MFI") + xlab("") +
  coord_flip() + scale_y_continuous(expand = c(0.01,0)) + 
  theme_bw()

# code below from https://wilkelab.org/cowplot/articles/plot_grid.html
# plotting both
plot_row <- plot_grid(p1, p2, rel_widths = c(1,1.3), labels = "AUTO")

# now add the title
title <- ggdraw() +
  draw_label(
    "GFP fluorescent intensity (arbitrary units), dashed line indicates background fluorescence",
    fontface = 'bold',
    x = 0,
    hjust = 0
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )
plot_grid(
  title, plot_row,
  ncol = 1,
  # rel_heights values control vertical title margins
  rel_heights = c(0.07, 1)
)
```
**Figure 1 Pho4-GFP intensity by strain.** Pho4-GFP intensities were quantified on an Attune NxT flow cytometer using 400 mV on the BL1 channel. Acquisition rate is 200 ul/min. Strain IDs are listed on the vertical axis while the Median Fluorescent Instensity (MFI, arbitrary units) on the horizontal axis. An asterisk next to a bar indicates an overnight growth instead of a mid-log phase growth was used for quantification. **(A)** omits A3, A5 and A6, which appear to be outliers to the rest. C2, B3 and D1 also appear to be much brighter than the rest. **(B)** includes all samples. Notice how the scale is dominated by A3,5,6.

_Discussion_

- With limited samples for the endogenously integrated Pho4, it appears that this method generates less clone-to-clone variability.
- Comparing G and H, which are endogenously integrated with mNeon or EGFP tag, it is clear that the former is brighter. The same comparison with the exogenously integrated strains, e.g. A vs B or D vs E, gave less clear answers. Specifically, B1-3 appear to show higher intensity than A's, although a couple of A clones, i.e. A3, A5 and A6 are outliers as listed above. Also, in my previous test on 2020-11-09, B clones are generally dimmer than A clones. So there may be some reproducibility issues.
- The outliers mentioned above are all within the exogenous group. My suspicion is that the ends-in integration method by the pRS305 based plasmids are likely to result in multiple integrations, which could happen at higher rate when a larger amount of digested plasmid is used in the transformation.
- The plasmid-borne Pho4-GFP strains didn't grow to saturation as the others did, likely due to expression levels of the Leu2 gene from the plasmids. As a result, for most of them I used the o/n culture. The two mid-log phase samples, M1 and M2, appear to be brighter than the o/n samples, although one needs to repeat this to know for sure.

Next let's see the population heterogeneity within each strain, using the percent robust CV measure offered by the Attune software.
```{r plot_robust_CV, fig.height = 6, fig.width = 6}
# let's also check the "spread" using %CV or %robustCV to see which methods produce more heterogeneity among cells
dat %>% filter(Parameter == "Pho4-GFP") %>% 
  ggplot(aes(x = Sample, y = rCV, color =  Group)) + 
  geom_point() +
  #geom_bar(stat = "identity", position = position_dodge2(width = 0.9)) +
  geom_text(aes(label = ifelse(Growth == "ovn", "*", ""), hjust = 1.2, vjust = 0.8), 
            size = 5, color = "white") +
  facet_grid(Pho4GFP~., scales = "free_y", space = "free_y") +
  scale_fill_brewer(type = "div") + ylab("%rCV") +
  coord_flip() + scale_y_continuous(expand = c(0.01,0)) +
  labs(title = "%rCV for Pho4-GFP fluorescent intensity", subtitle = "Asterisks indicate o/n instead of mid-log phase culture")
```
**Figure 2 Pho4-GFP fluorescence cell-to-cell variability within each strain.** Pho4-GFP intensities were quantified on an Attune NxT flow cytometer using 400 mV on the BL1 channel. Acquisition rate is 200 ul/min. Strain IDs are listed on the vertical axis while the percent robust CV (%rCV) on the horizontal axis. An asterisk next to a bar indicates an overnight growth instead of a mid-log phase growth was used for quantification.

_Discussion_

- P1-4 had the highest level of spread, although it is unclear if that's due to the o/n growth or due to the strain itself.
- Note that the outliers in GFP fluorescence in **Fig. 1**, which are exogenously integrated, _don't_ have higher variability, suggesting that they are stable, but perhaps multiple integrants.

### PHO5pr-mCherry raw intensities

```{r, fig.height = 6, fig.width = 7}
# calculate rfp background and basal expression level
rfp.bg <- mean(dat %>% filter(!PHO5RFP, Parameter == "PHO5pr-mCherry") %>% pull(Median))
rfp.basal <- dat %>% filter(PHO5RFP, Pho4 == "pho4-", Parameter == "PHO5pr-mCherry") %>% pull(Median) %>% mean() - rfp.bg

p1 <- dat %>% filter(PHO5RFP) %>% 
  ggplot(aes(x = Sample, y = Median, fill =  Group, color = PHO2)) + 
  geom_bar(stat = "identity", position = position_dodge2(width = 0.9)) +
  geom_hline(yintercept = rfp.basal, linetype = 2, alpha = 0.6) +
  geom_text(aes(label = ifelse(Growth == "ovn", "*", ""), hjust = -0.2, vjust = 0.8), color = "black", size = 5) +
  facet_grid(Pho4 ~ Parameter, scales = "free", space = "free_y") + ylab("MFI") +
  scale_fill_brewer(type = "div") + scale_color_manual(values = c("red", NA)) +
  coord_flip() + scale_y_continuous(expand = c(0.01,0.02)) +
  theme_bw()
p1
```
**Figure 3 Corresponding Pho4-GFP and PHO5pr-mCherry reporter fluorescence intensity per strain.** Pho4-GFP and PHO5p-mCherry intensities were quantified on an Attune NxT flow cytometer using 400 mV on the BL1 and 430 mV on the YL2 channels respectively. Acquisition rate is 200 ul/min. At least 10,000 events were collected. Strain IDs are listed on the vertical axis and the Median Fluorescent Intensity (MFI, arbitury units) for each channel on the horizontal axis. An asterisk next to a bar indicates an overnight growth instead of a mid-log phase growth was used for quantification. The dashed vertical line for Pho4-GFP indicates the autofluorescence as calculated by the mean of the intensities in the strains without GFP; the dashed vertical line for the RFP plot indicates the basal level expression from the reporter without Pho4, calculated by the average of yH295 and yH296, both of which lack Pho4 but has the reporter).

_Discussion_

- CgPho4 clearly has no dependence on Pho2 when assayed in _S. cerevisiae_, largely consistent with my previous results (although my RNA-seq has indicated a modest dependence with ~30-40% reduction in mRNA levels of _PHO5_ driven by CgPho4 without Pho2.)
- ScPho4 has strong dependence on Pho2. P1-4 and B1,2,4 both showed that without Pho2, ScPho4 doesn't induce reporter expression on its own. Some anormalies exist however, as will be discussed below.
- H1-3 are supposed to be ScPho4-EGFP with the PHO5pr-mCherry and also with ScPho2. However, the results here suggest that this strain had little reporter induction. It may be that the integrated ScPho4-EGFP, even though it is expressed, is not functional. Is this consistent with Lindsey's results, including flow cytometry and liquid phosphatase assays?
- G2, 3, 4 have much higher levels than everyone else. Given that the G strains were constructed by Lindsey by integrating the pRS306-PHO5pr-mCherry-URA3 construct into the ura3 locus using the ends-in transformation, I wonder if G2,3 and 4, just like those leu2 integrated Pho4 variants, represent multiple integrations.
- Lastly, note that the several A clones with high Pho4-GFP (A3, A5 and A6) also showed higher levels of reporter expression, suggesting that increasing the dose of Pho4 can certainly partially compensate for the loss of Pho2.

### _PHO5pr-mCherry_ induction levels normalized by Pho4 abundance
We will first transform the raw GFP and RFP intensities to make them more interpretable. For the GFP intensity, as there is a substantial background, we will subtract the background from all the GFP-containing strains to obtain the meaningful measure for Pho4 protein levels. For RFP, there is minimal background (see below). However, there is a "basal" expression of the reporter in the absence of Pho4 -- what we are interested in is not the absolute level of reporter expression but the fold induction compared with the _pho4∆_ strains.
```{r transform}
# calculate gfp background
gfp.bg <- dat %>% filter(Pho4GFP == "No-GFP", Parameter == "Pho4-GFP") %>% pull(Median) %>% mean()
# transform
dat1 <- dat %>% select(-Count, -Mean, -CV, -rCV, -Percent) %>% 
  pivot_wider(names_from = Parameter, values_from = Median) %>% 
  mutate(GFP.noBG = ifelse(Pho4GFP == "No-GFP", NA, `Pho4-GFP` - gfp.bg),
         RFP.noBG = ifelse(PHO5RFP, `PHO5pr-mCherry` - rfp.bg, 0),
         RFP.FC = ifelse(PHO5RFP, RFP.noBG / rfp.basal, NA),
         nRFP.FC = RFP.FC / GFP.noBG * median(GFP.noBG, na.rm = T))
```

```{r plot_normalized_induction, fig.height = 7, fig.width=10, warning=FALSE}
dat2 <- dat1 %>% filter(Pho4GFP != "No-GFP", PHO5RFP)
p1 <- dat2 %>% 
  mutate(GFP.noBG = GFP.noBG / 1000, RFP.noBG = RFP.noBG / 1000) %>% 
  pivot_longer(cols = c(GFP.noBG, RFP.noBG), names_to = "Parameter", values_to = "Value") %>%
  mutate(Parameter = factor(Parameter, levels = c("GFP.noBG", "RFP.noBG"),  
                            labels = c("Pho4-GFP abundance", "PHO5pr-mCherry level")),
         Group = factor(Group, levels = c("control", "endogenous", "exogenous", "plasmid"))) %>% 
  ggplot(aes(x = Sample, y = Value, fill =  Group, color = PHO2)) + 
  geom_bar(stat = "identity", position = position_dodge2(width = 0.9)) +
  geom_text(aes(label = ifelse(Growth == "ovn", "*", ""), hjust = -0.2, vjust = 0.8), color = "black", size = 5) +
  facet_grid(Pho4GFP ~ Parameter, scales = "free", space = "free_y") +
  scale_fill_brewer(type = "div", drop = FALSE) + scale_color_manual(values = c("red", NA)) +
  coord_flip() + scale_y_continuous(expand = c(0.01,0.02), trans = "sqrt") + ylab("MFI (x1000, a.u.)") + 
  theme_bw()
  
p2 <- dat2 %>% 
  pivot_longer(cols = c(RFP.FC, nRFP.FC), names_to = "Parameter", values_to = "Value") %>%
  mutate(Parameter = factor(Parameter, levels = c("RFP.FC", "nRFP.FC"), 
                            labels = c("PHO5 fold induction", "PHO5 induction / Pho4-GFP")),
         Group = factor(Group, levels = c("control", "endogenous", "exogenous", "plasmid"))) %>% 
  ggplot(aes(x = Sample, y = Value, fill =  Group, color = PHO2)) + 
  geom_bar(stat = "identity", position = position_dodge2(width = 0.9)) +
  geom_hline(yintercept = 1, linetype = 2) +
  geom_text(aes(label = ifelse(Growth == "ovn", "*", ""), hjust = -0.2, vjust = 0.8), color = "black", size = 5) +
  facet_grid(Pho4GFP ~ Parameter, scales = "free", space = "free_y") +
  scale_fill_brewer(type = "div", drop = F) + scale_color_manual(values = c("red", NA)) +
  coord_flip() + scale_y_continuous(expand = c(0.01,0.02), trans = "sqrt") + ylab("Fold Induction") + xlab("") +
  theme_bw()

## place shared legend at the bottom
## reference: https://wilkelab.org/cowplot/articles/shared_legends.html
legend <- get_legend(
  p1 + 
  guides(color = guide_legend(nrow = 1), fill = guide_legend(nrow = 1)) +
  theme(legend.position = "bottom")
)
prow <- plot_grid(
  p1 + theme(legend.position="none"),
  p2 + theme(legend.position="none"),
  align = 'vh',
  labels = c("A", "B", "C"),
  hjust = -1,
  nrow = 1
)
plot_grid(prow, legend, ncol = 1, rel_heights = c(1, .1))
```
**Figure 4 Composite plot including Pho4 abundance, _PHO5_ reporter strength and fold induction values.** Pho4-GFP and PHO5p-mCherry intensities were quantified as defined above. Strain IDs are listed on the vertical axis. In **(A)**, the x-axis values represent the Median Fluorescent Intensity (MFI, arbitury units) for either Pho4-GFP or PHO5pr-mCherry after subtracting the background (based on strains without the corresponding fluorescent protein). In **(B)**, the fold change for _PHO5_ reporter is calculated as the background-subtracted RFP level divided by the basal expression level, as measured by the mean of the two strains containing the reporter but have _pho4∆_. The normalized induction ratio on the right column is calculated by dividing the induction fold change from the left column by the corresponding Pho4-GFP (background subtracted) levels and multiplied by the median Pho4-GFP levels of all strains. The dashed lines in both columns indicate the induction fold change of 1, i.e. no change compared to the _pho4∆_ strains. In both panels, the x-axes are square-root transformed so as to bring outliers into the plot without compressing the rest of the data range. An asterisk next to a bar indicates an overnight growth instead of a mid-log phase growth was used for quantification.

_Discussions_

- If we look horizontally across the four columns for each alphabetic group (A, B, C, ..., P), we can see that the plasmid born Pho4's actually behave pretty well, especially after normalizing the RFP induction fold change by the Pho4 abundance.
- Next, the endogenously integrated ScPho4-mNeon showed small clone-to-clone variability in the GFP level, but showed huge variation in the _PHO5pr-mCherry_ levels. The normalized induction fold changes continue to show the dramatic differences between clones. As mentioned before, we suspect this is due to the multi-integration of the reporter construct at the ura3 locus.
- The exogenously integrated ScPho4-mNeon exhibited large clone-to-clone variability in the GFP level, again possibly due to multi-integration of the Pho4-GFP in the _leu2_ locus. The normalization by Pho4 abundance didn't fully remove the difference, likely because the induction fold change doesn't scale linearly with Pho4 levels (saturation effect).
- ScPho4-EGFP, both the endogenously and exogenously integrated ones, had issues that prevent a clear interpretation.
- CgPho4-mNeon is similar to ScPho4-mNeon in that it showed relatively consistent GFP levels except for C2, which may be multi-integration. The normalized induction fold changes are pretty consistent with or without Pho2, except for C2, which showed a lower value likely indicating non-linear relationships between induction and Pho4 abundance.

## Conclusions
- mNeon appears to be as good as, if not better than EGFP in terms of both brightness and keeping the tagged Pho4 functional.
- the ends-in integration seems to generate multiple integrations frequently (maybe a result of excess digested plasmid? but we are also fighting relatively low transformation efficiency)
- increasing the dose of ScPho4 can partially compensate for the loss of Pho2, which further highlights the importance of ensuring equal or highly similar levels of expression for all constructs to be tested.