---
title: "Ex009_main_analysis"
author: Bin He
created: 19 oct 2017
output: 
  html_notebook:
    toc: true
    toc_float: true
---

```{r global_options, include=FALSE}
rm(list=ls()) ### To clear namespace
library(knitr)
opts_chunk$set(echo=TRUE, warning=FALSE, message=FALSE, fig.path="../output/figure")
```

## Goal

1. Characterize the **genome-wide** transcriptional response to inorganic phosphate starvation in _C. glabrata_. Compare this to the Pho4-dependent gene induction identified in the _pho80∆_ background that causes constitutive activation of Pho4.
    
    > previously I used RNAseq in _pho80∆_ to identify genes induced by Pho4, while measuring Pho4 binding profile under physiological starvation conditions. This dataset will allow me to revisit some of the genes deemed bound but not induced by Pho4 in the previous dataset, to see if they are actually induced under starvation.

1. in addition to identifying the genes that respond to the starvation, but also their kinetics.
    
    > this is motivated by Barkai group's Cell Reports study, where they showed that PHO genes in _S. cerevisiae_ were induced in two big waves following starvation. I want to see if similar dynamics can be observed in _C. glabrata_
   
## Experiment set up

In Ex009, I performed time-course transcriptome profiling following inorganic phosphate starvation in two genetic background, namely the wild-type and _pho4∆_. The purpose of doing this in _pho4∆_ is to distinguish Pho4-dependent vs Pho4-independent gene induction.

The time-course includes a pre-stress (rich media) condition and a series of post-stress time points. The sampling density on the time axis is denser in the beginning, then gradually becoming sparser, in contrast to the regular interval designed used in Barkai's study. This is so that I can focus on the early time points while restricting the total number of samples I need to collect and process.

The sample sheet is documented [here](../data/sample_sheet/Ex009_experiment_set_up_20171019.csv)

Below is a simple chart to show the design:

| Genotype | Time point                | Replicates | Comment |
| -------- | :------------------------ | :--------: | ------- |
| 80∆      | rich media                | 2          | for consistency check with my previous data |
| 80∆ 4∆   | rich media                | 2          | for consistency check with my previous data |
| wt-1     | pre, 20', 30', 45', 60', 90', 120', 150', 180', 240' | 1 | time course for wt  |
| 4∆-1     | pre, 20', 30', 45', 60', 90', 120', 150', 180', 240' | 1 | time course for 4∆  | 
| wt-2     | pre, 20', 30', 45', 60',      120',       180', 240' | 1 | biol. repl. for wt-1|
| 4∆-2     | pre, 20', 30', 45', 60',      120',       180'       | 1 | biol. repl. for 4∆-1| 

## Analyses

### Prepare data

```{r load_libraries, message=FALSE}
require(ggplot2)       ## for plotting
require(cowplot)       ## for some useful defaults in ggplot2
require(limma)         ## for differential gene expression analysis
require(data.table)    ## for fast importing and manipulating data tables
require(edgeR)         ## for DGEList()
require(RColorBrewer)  ## for plotting
require(NMF)           ## for heatmap plot
source("../code/R_subfunctions_20171026.R")  ## custom defined functions
```

```{r load_data}
# raw data count
raw <- fread("../data/gene_count/Ex009_reads_per_transcript_2017-10-18.txt", key = "gene.names")
# sample sheet
sample <- fread("../data/sample_sheet/Ex009_experiment_set_up_20171019.csv")
sample[,Group:=paste(Genotype,Replicate,Timepoint, sep="-")]
sample[,Timepoint:=factor(Timepoint, levels = c("pre","20m","30m","45m","60m","90m","120m","150m","180m","240m","del80"))]
# rearrange the columns in `raw` according to the samples order in `sample`
ind <- match( c(sample$Sample, "gene.names"), names(raw) )
raw <- raw[,ind,with=FALSE]
# annotation
anno.file <- fread("../data/annotation/C_glabrata_gene_for_mapping_s02-m07-r04.bed")
names(anno.file) <- c("Chr","Start","End","GeneID","Not.Use","Strand","GeneName","Type.of.Gene")
anno <- anno.file[,c("Chr","GeneID","GeneName","Type.of.Gene")]
setkey(anno, "GeneID")
```

```{r prepare_data}
# filter dataset to remove very lowly expressed genes
# 1. examine the distribution of reads for each gene across all samples, to establish a threshold
S <- rowSums(raw[,1:40])
plot(density(log10(S)))
sprintf("The number of genes with less than 40 total counts across 40 samples, i.e. 1 read per sample on average is %d, and those with less than 100 total counts, or 2.5 reads per sample, is %d", sum(S<=40), sum(S<=100))
print("Let's try 40 reads as a cutoff for dropping genes with low or no expression")

# 2. filter dataset
isexpr <- (S <= 40)
use.genes <- grepl("ncRNA|ORF|pseudogene", anno$Type.of.Gene)
use <- (!isexpr & use.genes)
mat <- as.matrix(raw[use, 1:40])
anno.all <- anno; anno <- anno.all[use]

# 3. write out the filtered table (uncomment if changes are made to the filtering steps above)
#write.table(as.data.frame(raw[use]), file = paste("../data/gene_count/Ex009_reads_per_transcript_filtered_", Sys.Date(), ".txt", sep = ""), quote = FALSE, row.names = FALSE)
```

### Exploratary analysis

_Goal_

- visualize the between sample variation in the raw read count distribution, so as to assess the need of normalization and to choose the appropriate method
- examine variability between technical and biological replicates, and separation between treatment, on an MDS plot

_Reference_

[RUVSeq manual on Bioconductor](http://bioconductor.org/packages/release/bioc/vignettes/RUVSeq/inst/doc/RUVSeq.pdf)

_Analyses_

1. Explore different normalization methods
```{r explore_normalization, fig.width=9, fig.height=3, fig.keep=TRUE}
require(aroma.light)        ## for exploratory analysis
# visualize the between sample variation pre-normalization
print("Non-normalized")
plotRLE(mat, outline=FALSE, ylim= c(-3, 3), cex.axis = 0.8, las = 3)
title(main = "non-normalized", xlab = "sample", ylab = "log2 ratio of expression")

# try upper quantile normalization
print("Upper-quantile normalized")
set1 <- myBetweenLaneNormalization(mat, which="upper")
myPlotRLE(set1, outline=FALSE, ylim=c(-3, 3), cex.axis = 0.8, las = 3)
title(main = "upperquartile normalized", xlab = "sample", ylab = "log2 ratio of expression")

# try full quantile normalization
print("Full-quantile normalized")
set2 <- myBetweenLaneNormalization(mat, which="full")
myPlotRLE(set2, outline=FALSE, ylim=c(-3, 3), cex.axis = 0.8, las = 3)
title(main = "full quantile normalized", xlab = "sample", ylab = "log2 ratio of expression")

# try TMM normalization
# 1. construct count matrix
dge <- DGEList( counts = mat, genes = anno )
# 2. perform TMM normalization
dge <- calcNormFactors(dge)
# 3. calculate log transformed, normalized counts
cpm.o <- cpm( dge, normalized.lib.sizes = FALSE, log = FALSE, prior.count = 0.5) # unnormalized
cpm.n <- cpm( dge, normalized.lib.sizes = TRUE, log = FALSE, prior.count = 0.5 ) # normalized
myPlotRLE(cpm.o, outline=FALSE, ylim=c(-4, 4), cex.axis = 0.5, las = 3)
title(main = "library size normalized", xlab = "sample", ylab = "log2 ratio of expression")
myPlotRLE(cpm.n, outline=FALSE, ylim=c(-4, 4), cex.axis = 0.5, las = 3)
title(main = "TMM normalized", xlab = "sample", ylab = "log2 ratio of expression")
```

**Conclusion**

- All three methods tried, i.e. TMM, upperquartile and full quantile normalization effectively homogenizes the distribution of log expression ratios.

- The distribution of log2 radio of expression is greater for S1-S6, S37 and S38. What they have in common is that all of them were grown in rich (non-Pi-starved) media. S37 and S38 mimics Pi starvation in that it also activates Pho4, which is supposed to turn on its targets, but as my analysis below shows, the majority of the transcriptional responses are likely to be Pho4-independent and thus it is not surprising that S37 and S38 look more like S1-6.

- I also looked at other properties of the samples and sequencing stat, such as the amount of RNA extracted, number of PCR cycles used for making the library, total number of reads obtained and number of reads mapped. None suggested quality differences between S1-6, S37, S38 with the rest samples.


```{r normalize_data}
# 1. construct count matrix
dge <- DGEList( counts = mat, genes = anno )
# 2. perform TMM normalization
dge <- calcNormFactors(dge)
# 3. use TMM to normalize data
lcpm <- cpm( dge, normalized.lib.sizes = TRUE, log = TRUE, prior.count = 0.5)
rownames(lcpm) <- anno$GeneID
```

2. MDS plot
```{r MDS_plot, fig.width=10, fig.height=5, echo=FALSE}
# the goal here is to perform unsupervised clustering to project the different samples, including biological and technical replicates, on a lower dimention that capture most of the variation
tp <- factor(sample[match(Sample, colnames(lcpm)), Timepoint])
gp <- factor(sample[match(Sample, colnames(lcpm)), paste(Genotype, ifelse(Timepoint %in% c("del80","pre"), "happy","stress"), sep=".")])
col <- brewer.pal(nlevels(gp), name = "Set1")
layout(matrix(c(1,2),ncol=2))
plotMDS(lcpm, labels = tp, col = col[gp], dim.plot = c(1,2), top = 500)
legend("topleft", legend = levels(gp), text.col = col)
title(main = "Top 500 differentially expressed genes")
plotMDS(lcpm, labels = tp, col = col[gp], dim.plot = c(1,2), top = 50)
legend("topleft", legend = levels(gp), text.col = col)
title(main = "Top 50 differentially expressed genes")
```

**Conclusion**

- The first dimension (leading logFC dim 1), which explains the most variation, mainly separates the non-stressed samples (including del80.del4) and the stressed ones. The del80.PHO4 sample is only slightly removed from the unstressed ones suggesting that Pho4-dependent gene induction probably represents a rather small fraction of the genes whose expression change upon stress.

- On the left graph, which used top 500 differentially expressed genes (see manual of plotMDS for details), we see that the stressed samples nicely align themselves in the order that largely match their time under stress (from top to bottom: 30'm 45', 60', 90', 120-240'). The del4 and PHO4 samples with matching time points group together, again suggesting that the majority of transcriptional responses after Pi starvation are Pho4-independent.

- I reasoned that the more strongly a gene is induced, the more likely it is under direct Pho4 regulation. Based on this assumption, I used the top 50 genes to construct the MDS plot (right). Here it indeed changed some of the features. Specifically, 

    1. the first dimention still mainly separates the non-stressed from the stressed samples. This time however, it also displays a gradient that matches the time under stress. 
    1. More interestingly, the second dimension now separates the del4 vs PHO4-wt samples. Moreover, it unexpected spreads the different time point samples, but only for del4 ones. The meaning of this is yet unclear.

3. Single gene analysis

_Goal_

- Visualize a few known Pho4 target genes and compare them to my expectation.

_Approach_

I wrote a sub-function in a separate file and source it here.

```{r single_gene_analysis, fig.width=12, fig.height=10}
set1 <- c("CAGL0B02475g", "CAGL0A01243g", "CAGL0L05456g", "CAGL0C02321g", "CAGL0G06952g", "CAGL0L06622g", "CAGL0M12705g", "CAGL0F02145g", "CAGL0J07040g", "CAGL0K12034g") # conserved targets with S. cerevisiae
myGenePlot(genes = set1)

set1.1 <- c("CAGL0K07546g", "CAGL0M11660g", "CAGL0B02453g", "CAGL0F02387g", "CAGL0K07524g", "CAGL0M12430g") # Cg. specific PHO related
myGenePlot(genes = set1.1)

set2 <- c("CAGL0E05588g", "CAGL0C04741g", "CAGL0M13189g", "CAGL0E05566g", "CAGL0E05984g", "CAGL0M11660g", "CAGL0G01540g", "CAGL0K10604g", "CAGL0J04202g", "CAGL0L06644g", "CAGL0K10164g", "CAGL0M06325g") # other stress related
myGenePlot(genes = set2)

set3 <- c("CAGL0J04202g", "CAGL0M07634g", "CAGL0M12430g", "CAGL0E05940g", "CAGL0K10164g", "CAGL0L06424g", "CAGL0L07502g", "CAGL0M08514g", "CAGL0J06050g", "CAGL0F00649g", "CAGL0M06325g", "CAGL0C04741g", "CAGL0B02926g", "CAGL0B02970g", "CAGL0I00726g") # cell wall and adhesion
myGenePlot(genes = set3)
```

### Compare pho80∆ gene list with previous results

_Goal_

Check if the gene list under pho80∆ background in this experiment is consistent with previous result

_Steps_

1. Load raw counts from previous result.
1. Extract the subset of data to be used.
1. Filter the data using the same gene list as in this analysis.
1. Normalize the data using the same method, but separately for the 2013 data.
1. Apply LIMMA-voom to identify top gene lists

_Analyses_

```{r prepare_2013_data, fig.width=10, fig.height=5}
# I. Load data
raw13 <- fread("../data/gene_count/2013_Cgla_reads_per_transcript_2016-02-21.txt") # checked to make sure that the gene names align with raw
sample13 <- fread("../data/sample_sheet/2013_CglaRNAseq_sample_info.csv")
# what I need is sample N1, N2, O1, O2, which correspond to S37, S38, S1, S2
# also use the same gene filtering I used for the current analysis
mat13 <- as.matrix( raw13[use, .(N1,N2,O1,O2)] )

# II. now combine the two experiments into one
mat.d80 <- cbind( mat[,c("S37","S38","S1","S2")], mat13 )
# and create the appropriate sample sheet
sample.d80 <- read.table(strip.white = TRUE, text = "Sample, Pho4, Exp
                         S37, Cgla4, 2017
                         S38, Cgla4, 2017
                         S1, del4, 2017
                         S2, del4, 2017
                         N1, Cgla4, 2013
                         N2, Cgla4, 2013
                         O1, del4, 2013
                         O2, del4, 2013", sep = ",", head = TRUE )

# III. apply TPP normalization
# 0. visualize the between sample unwanted variation using Relative Log Expression (RLE)
oldpar <- par( mfcol = c(1,2) )
myPlotRLE( mat.d80, outline = FALSE, ylim = c(-2.3,2.3), las = 2, main = "RLE plot before normalization" )
# 1. construct count matrix
dge.d80 <- DGEList( counts = mat.d80, genes = anno )
# 2. perform TMM normalization
dge.d80 <- calcNormFactors(dge.d80)
# 3. use TMM to normalize data
cpm.d80 <- cpm( dge.d80, normalized.lib.sizes = TRUE, log = FALSE, prior.count = 0.5)
# 4. examine the RLE plot after normalization
myPlotRLE( cpm.d80, outline = FALSE, ylim = c(-2.3,2.3), las = 2, main = "RLE plot after normalization" )
# 5. finish the normalization with log transformation
lcpm.d80 <- cpm( dge.d80, normalized.lib.sizes = TRUE, log = TRUE, prior.count = 0.5)
rownames(lcpm.d80) <- anno$GeneID

# IV. bird-eye view of similarity between samples
plotMDS( lcpm.d80, top = 500, main = "MDS plot with top 500 DE genes")
plotMDS( lcpm.d80, top = 50,  main = "MDS plot with top 50  DE genes")
par( oldpar )
```

```{r del80_DE_analysis, fig.height=10, fig.width=10, fig.keep=TRUE}
layout(matrix(c(1,2,0,3), ncol = 2))
# the goal here is to perform differential gene expression analysis, separately for 2013 and 2017 data
# create design matrix
grp <- with(sample.d80, paste(Pho4,Exp,sep="."))
dsgn.d80 <- model.matrix(~0+grp)
colnames(dsgn.d80) <- gsub("grp","",colnames(dsgn.d80))
# voom transformation to account for variance-mean dependence
v.dge.d80 <- voom(dge.d80, design = dsgn.d80, plot = TRUE)
# v.dge.d80 <- voomWithQualityWeights(dge.d80, design = dsgn.d80, plot = TRUE) # incorporate sample weights

# make contrasts
contr.d80 <- makeContrasts(
  Pho4.13 = Cgla4.2013 - del4.2013,
  Pho4.17 = Cgla4.2017 - del4.2017,
  Cgla4.13vs17 = Cgla4.2013 - Cgla4.2017,
  del4.13vs17 = del4.2013 - del4.2017,
  levels  = dsgn.d80
)

# LIMMA fit
fit.d80 <- lmFit(v.dge.d80, design = dsgn.d80) # this automatically extracts the weights in the v.dge.d80 and uses it to perform a weighted linear regression
fit.d80 <- contrasts.fit(fit.d80, contrasts = contr.d80)
fit.d80 <- eBayes( fit.d80 )
sig.d80 <- decideTests( fit.d80, method = "global", lfc = 1 ) # only include genes more than 2 fold induced
choose <- rowSums(sig.d80[,1:2]) >= 1 # select genes that are deemed significant in at least one exp.
test <- coef(fit.d80)[choose,]
lm <- lm(test[,"Pho4.13"]~test[,"Pho4.17"])

# plot to compare the two results
plot(coef(fit.d80)[choose,1:2], cex = 0.75, pch = 19, col = rgb(0,0,0,0.6), xlab = "2013", ylab = "2017", main = "log2 fold-change", sub = paste("Pearson's r = ", round(cor(test[,1],test[,2]),2) ,sep="") )

# plot the overlap between the two list
vennDiagram( sig.d80[,1:2], include = "up", circle.col = c("turquoise","salmon") )

# finally, just list the top genes
print(topTable( fit.d80, n = 100 ))
```

```{r del80_plot_heatmap, fig.width=8, fig.height=10}
sprintf("A totle of %d genes were selected because they were significantly induced by more than 2 fold in at least one of the two experiments.", sum(choose))
mycol <- colorpanel(1000,"blue","black","yellow")
heatmap.2(v.dge.d80$E, scale="row",Rowv = TRUE,
   labRow=NA, labCol=colnames(v.dge.d80),
   col=mycol, trace="none", density.info="none", 
   margin=c(8,6), lhei=c(2,10), dendrogram="column")
```