---
title: RNAseq data for Ex009, Pi starvation time course in _C. glabrata_
author: Bin He
date: 10 oct 2017
---

# Sample

This folder contains raw sequencing data and processed data from the experiment Ex009
40 samples pooled into 2 pools, using 20 adapter indices for each 

# Notes

## [18 oct 2017] Gene counting with bedtools, summary (finish work on the server's side)

- bedtools is installed on the head node, version v2.23.0

	- I would like to install v2.25.0 or newer to take advantage of the `-sorted` option, which both speeds up the process and also outputs results in the order given in the annotation file
	- After consulting with csgenomes, I was able to use 2.26.0 by calling `/usr/local/BEDTools/2.26.0/bin/bedtools`
	- I need to make a few modification on the command-line, but nothing major.
	- Resulting output files were then processed by the `Post_bedtools_analysis.R` script, which I also modified to fit the current need.
	- Manually checked _PHO84_ and _PMU2_ induction levels. Results largely are as expected, although there appears to be a very low level of induction in _$pho4 \Delta$_ strains, which is not expected.
	- Sync the results to my local folder for further analysis. 

- 
## [17 oct 2017] IGV browser, look at raw data

- managed to get x11 forwarding working

	- initially I got the "error locking $HOME/.Xauthority" error. after googling, I understand that this file must have 600 permission (read/write only by user). But mine is correct. Then I found there is another `.Xauthority-c` file. After moving it to `.Xauthority-c.bk` and reconnecting with `ssh -X gen-comp1`, x11 forwarding is now working. Tested with `xterm`
	- downloaded the latest igv browser from [broad institute](http://software.broadinstitute.org/software/igv/LoadGenome) and unzipped the folder under `$HOME/src`
	- created hard links for `igv.sh` and `igv.jar` in the `$HOME/bin` directory, and successfully launched the app
	- launched IGV and loaded the genome file I made before under `$HOME/data/NGS/Genome_Annotation/C_glabrata`
	- the GUI is very sluggish, barely usable

## [13 oct 2017] Bowtie mapping

_Goal_

- Perform bowtie mapping for all 40 samples against the latest Cgla genome

_Notes_

1. bowtie 0.12.7 is installed on gen-comp1. bowtie2 2.2.0 is also installed. According to the software's developers ["bowtie2 is not a drop-in replacement for bowtie 1"](http://bowtie-bio.sourceforge.net/bowtie2/faq.shtml). I decided to use bowtie 0.12.7 first, as all the parameters I used before work.

1. `samtools sort` often fails in my previous experience. after looking at the [documentation](http://www.htslib.org/doc/samtools-1.3.html), I figured out the problem: use of `-T /tmp/sort_tmp` flag together with submitting array jobs (multiple jobs at the same time) can create conflict between the different jobs -- one job may have just deleted the tmp file it created when another wants to access it. Solution is rather simple: just leave out the flag and let samtools use the default, which is to create temporary files along side the `sam` file. Also, use of `-m 10G` will allocate 10G of memory towards each task such that it can hold enough data in the memory and can thus skip the merging step.

_Results_

- average uniquely mapped reads percentage is ~90%. The lowest (S10) is 82%

## [12 oct 2017] Demultiplex

- To demultiplex, one needs to go to the individual assay page, find the button that says "create custom barcodes", and follow the instructions.

- Pool A is sequenced together with 5 other samples. The i7 adapter, which contains the TruSeq index, conflicts between my sample and those 5 samples. Wang Wei at the core pointed out that my sample and the other person's differ in the i5 index, which is read as Read 3, and can be used to distinguish our samples from each other. My i5 index is TCTTTCCC, while the other party's is GGTTGAGA.

- The HTSEQ server most likely uses [bcl2fastq2](https://support.illumina.com/content/dam/illumina-support/documents/downloads/software/bcl2fastq/bcl2fastq2-v2-18-software-guide-15051736-01.pdf) from illumina to do the demultiplexing. This software has a flag `--barcode-mismatches` that can be set to 0, 1 or 2, with the default being 1. Wei suggested using 2 for both i5 and i7 adapters.

- Some thoughts on the mismatches allowed:

	Calculate hamming distance between TruSeq adapter indices 1-20

	```{r test}
	require(stringdist)
	seq <- c("ATCACG", "CGATGT", "TTAGGC", "TGACCA", "ACAGTG", "GCCAAT", "CAGATC", "ACTTGA", "GATCAG", "TAGCTT", "GGCTAC", "CTTGTA", "AGTCAA", "AGTTCC", "ATGTCA", "CCGTCC", "GTCCGC", "GTGAAA", "GTGGCC", "GTTTCG")
	dist <- stringdistmatrix(seq, seq, method = "hamming")
	print("The distribution of pairwise hamming distance is:")
	print(table(dist))
	print("Note that the 20 zeros are self-distance")
	```
	
	output

	```
	dist
	0   3   4   5   6 
	20  56  98 166  60
	```
	
	It is important to realize that allowing more mismatches won't dramatically inflate the rate of index misassignment, i.e. read index 1 as index 2. This is because any index read that is equal distance away from the supplied set of indices used will be discarded.

- Results of demultiplex

    - PoolA is co-sequenced with 5 other samples that share the i7 barcodes with 3 of my samples. As a result, I need to utilize the i5 adapter sequence, which was in Read 3. This dual index demultiplexing is more complex and resulted in a higher percentage of unmatched data. When allowing 1 mismatch (on each of i5 and i7 index), ~35% of the reads are dropped into the "unmatched" bin. When allowing 2 mismatches, ~25% fall in the "unmatched" bin.

    - On PoolB, which is only multiplexed with one other sample, I used a more stringent cutoff of 1 mismatch, and the percentage of unmatched reads (due to indices mismatch) is &lt 1%. Totally fine.

## [11 oct 2017] Sequencing completed

- Data available on HTSEQ.princeton.edu

- The HTSEQ website is essentially a front end for a MySQL database. It organizes all the sample information and run results using concepts including _Sample_, _Assay_, _Dataset_ etc. A sample is what users submit to the core. An assay refers to a run on the Illumina machine. Note that for the Rapid Flowcell run on HiSeq 2500, each run has two assays, although it would be the same pooled samples sequenced in both. They result in a "merged" fastq.
