---
title: "Ex009 library validation by PHO84 qPCR"
author: "Bin He"
date: '2017-09-29'
tags: Ex009
---

Summary
------------------
### Goal  

- Validate gene induction for timecourse RNAseq sample, Ex009

### Materials

#### Samples
 ID  | Sample            | Note
:----|:------------------|:-------------
 S1  | 80- 4-, high Pi   | no induction
 S3  | 80+ 4+, high Pi   | no induction
 S5  | 80+ 4-, high Pi   | no induction
 S21 | 80+ 4-, no Pi 60' | no induction
 S7  | 80+ 4+, no Pi 20' | induction
 S15 | 80+ 4+, no Pi 45' | induction
 S19 | 80+ 4+, no Pi 60' | induction
 S37 | 80- 4+, high Pi   | induction

#### Assays
  1. ACT1
  2. PHO84
  
##### qPCR plate design
  any comments on things such as dilution ratio for samples.

### Results

1. Pho84 is induced > 600 fold
2. gDNA contamination exists, but it's a very small amount, about 1-2% based on ACT1 and PHO84 estimates
3. DNase I treatment is very effective. It reduces the amount of gDNA below qPCR detection threshold in the no RT samples.

Setup Parameters
---------------------
```{r library_n_parameters, message=FALSE}
require(ggplot2)
require(cowplot)
#require(Hmisc)
require(plyr)
require(reshape2)
require(stringr)
```

Read & Format data
---------------------
```{r Read_data}
data <- read.table("../data/20170928_Ex009_batch_all_PHO84_validate.csv", sep=",", as.is=T, head=TRUE, fill=T, na.strings = "NA")
data$Ct <- as.numeric(data$Ct)
data$Sample <- factor(data$Sample, levels = c("NTC","S37","S7","S15","S19","S1","S3","S5","S21"))
data.abi <- subset(data, Master == "AmpliGold")
```

> **of note**, I arbitrarily set the Ct value for NTC to 42, so as to include them in the plots. In the assay, these reactions fail to generate amplicons, indicating absence of the suitable templates

```{r Data_overview_1, echo=FALSE, fig.width=9}
pd <- position_jitter(width=0.1)
p <- ggplot( na.omit(data.abi), aes(Sample, Ct, color = as.factor(Dilution), ymax=max(Ct)*1.05) )
print( p + geom_point(position = pd, shape = 24, size = 2) + facet_wrap( ~Assay) )
```

Summarize data
---------------------
### Standard curve and efficiency calculation
To calculate efficiency, use linear regression
$$ n_t = D_i n_0 E^{C_{T,i}} $$ 

```{r standard_curve_1, eval=TRUE, echo=FALSE, comment=NA, fig.width=9, warning=FALSE}
# Calc primer amplification efficiency
linmod <- function(df) {
  if( sum(!is.na(df$Ct)) == 0 ) { # if no non-NA data
    return( c("intercept"=NA, "slope"=NA, "p.value"=NA, "r.squared"=NA) )
  }
  lm <- lm( log2(Dilution) ~ Ct, data=df, na.action = "na.omit")
  intercept <- as.numeric(coef(lm)[1])
  slope <- as.numeric(coef(lm)[2])
  efficiency <- 2^(slope)
  sum <- summary(lm)
  p <- coef(sum)[2,4]
  R2 <- sum$r.squared
  return( c("efficiency"=efficiency, "intercept"=intercept, "slope"=slope, "p.value"=p, "r.squared"=R2) )
}
eff.abi <- ddply( subset(data.abi,Sample == "S37"), .(Assay), linmod)
eff.one <- ddply( subset(data,Sample == "S37" & Master == "OneTaq"), .(Assay), linmod)

# Standard curve, ABI AmpliGold master mix
print(eff.abi, digits=3)

p <- ggplot( subset(data.abi, Sample == "S37"), aes(Ct, log2(Dilution)) ) 
p <- p + geom_abline(data=eff.abi, aes(intercept=intercept, slope=slope))
p + geom_point() + facet_wrap( ~ Assay )

# Standard curve, Serge's OneTaq recipe
print(eff.one, digits=3)

p <- ggplot( subset(data, Sample == "S37" & Master == "OneTaq"), aes(Ct, log2(Dilution)) ) 
p <- p + geom_abline(data=eff.one, aes(intercept=intercept, slope=slope))
p + geom_point() + facet_wrap( ~ Assay )
```

In this experiment I also compared the commercial SYBR Green master mix from Applied Biosystems (AmpliGold) and Serge's recipe using NEB OneTaq. I used OneTaq only with one sample, S37, which was measured at four dilutions with both ABI and OneTaq. The amplification curves look better with the OneTaq. Otherwise the two are pretty much equivalent in terms of standard deviation and regression R^2

### Summarize Ct
```{r delta_Ct, echo=FALSE, comment=NA, fig.width=9, warning=FALSE}
## !! new code, match the replicate, form ratio, then calculate mean and sd !! ##
# 1. Mean and sd
res <- ddply( data.abi, .(Sample, Assay), summarize, 
              mean = mean(Ct-log2(Dilution), na.rm=T), 
              sem = sd(Ct-log2(Dilution), na.rm=T) / sqrt(length(Ct)), 
              .drop=TRUE )

# 2. Fold induction
## use dcast to make a table with ACT1 and PHO84 Ct values as columns
fold.d <- dcast(res, Sample ~ Assay, value.var = "mean")
## calculate dCt for PHO84 using ACT1 as control, within each Strain
fold.d$dCt <- fold.d$ACT1 - fold.d$PHO84
## calculate ddCt
ddCt <- with(fold.d, c( 
  "80-.4+.hiPi" = dCt[Sample == "S37"] - dCt[Sample == "S1"],
  "80+.4-.noPi.60'" = dCt[Sample == "S21"] - dCt[Sample == "S5"],
  "80+.4+.noPi.20'" = dCt[Sample == "S7"] - dCt[Sample == "S3"],
  "80+.4+.noPi.45'" = dCt[Sample == "S15"] - dCt[Sample == "S3"],
  "80+.4+.noPi.60'" = dCt[Sample == "S19"] - dCt[Sample == "S3"]
  ))
## print results
sprintf("Pho84 was induced %.0f fold in pho80∆ relative to pho80∆ pho4∆ under 10mM Pi", 2^ddCt["80-.4+.hiPi"])
t <- c(20,45,60)
sprintf("and %.0f fold in wildtype after %2d minutes of Pi starvation", 2^ddCt[paste0("80+.4+.noPi.",t,"'")], t)
sprintf("Pho84 was induced %.0f fold in pho4∆ after 60' of Pi starvation", 2^ddCt["80+.4-.noPi.60'"])
print("The last result was unexpected. Is this measurement error or is there an alternative way to induce PHO84 independent of Pho4?")
```

